src/vshaxe/display/HaxeDisplayArgumentsProvider.hx

```diff
@@ -197,13 +205,61 @@ class HaxeDisplayArgumentsProvider {
 
 		statusBarItem.hide();
 	}
+
+	function onDidChangeActiveTextEditor(editor:Null<TextEditor>) {
+		setConfigurationFromFilePath(editor != null ? editor.document : null);
+	}
+
+	function setConfigurationFromFilePath(document:Null<TextDocument>) {
+		if (document != null && document.languageId == "haxe") {
+			setFirstMatchingConfiguration(configurations.iterator(), document.uri.fsPath);
+		}
+	}
+
+	function setFirstMatchingConfiguration(configurations:ArrayIterator<Configuration>, documentFsPath:String) {
+		if (configurations.hasNext()) {
+			var config = configurations.next();
+			switch config.kind {
+				case Configured(idx, _, files) if (files != null):
+					matchFileWithPatterns(files.iterator(), documentFsPath).then((v) -> {
+						if (v) {
```

Comment:
One more thing, not a fan of single-letter var names like this. Should be `didMatch` or `matched` or something like that for better readability.